web/src/unread.js

```diff
@@ -283,9 +283,6 @@ class UnreadTopicCounter {
                     stream_count += topic_count;
                 }
 
-                // TODO: These don't agree with the else clause in how
-                // they handle muted streams/topics, and that's
-                // probably a bug.
```

Comment:
Hmm, the comment here is about how the `if/else` clauses here compute `res.stream_count` using different algorithms that don't agree; and that situation hasn't changed? Or am I missing something?

Reply:
I added a comment here. 
```
                // This should reflect the true unread count regardless of muted topics / streams
                // since inbox view uses it to decide whether to render the stream topics or not.
                // Note that the unread count here can be different than the unread count returned
                // in the else condition below.
```

Reply:
Since we are rendering everything in inbox and then hide them based on filters, the unread count here should actually be different than left sidebar.